src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -589,5 +638,95 @@ private static string GetFullPath(K8SConfiguration configuration, string path)
                 return Path.Combine(Path.GetDirectoryName(configuration.FileName), path);
             }
         }
+
+        private static void MergeKubeConfig(K8SConfiguration basek8SConfig, K8SConfiguration mergek8SConfig)
+        {
+            // For scalar values, prefer local values 
+            basek8SConfig.ApiVersion = basek8SConfig.ApiVersion ?? mergek8SConfig.ApiVersion;
+            basek8SConfig.CurrentContext = basek8SConfig.CurrentContext ?? mergek8SConfig.CurrentContext;
```

Comment:
This is tricky (and an example of why merge is hard) I'm not sure first or last config wins is what people expect.

Reply:
Isn't last wins most intuitive? (Like the PATH)

Reply:
According to https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#merging-kubeconfig-files:

> The first file to set a particular value or map key wins.

I think preferring first wins should be implied here. However, for specific values like CurrentContext, this may vary?

Reply:
Actually, I'm getting more confused now, because from what I can tell, [the go client is merging with overwrite for each kubeconfig](https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/loader.go#L229).

Reply:
Wait, first wins in the PATH 😄 ....

Reply:
Yeah, this is my broader point about fancy merges being mostly only documented in how it is implemented in kubectl.

We can emulate that or we can just throw when we hit something we don't understand/don't spec.

Reply:
Looking back, this is actually doing two passes with the merge, one forward and one backward. It still looks like first wins 😄